integration_tests/tokenuploadcontroller_test.go

```diff
@@ -109,7 +111,79 @@ var _ = Describe("TokenUploadController", func() {
 
 			})
 		})
+		When("when secret data are already present", func() {
+			var test crenv.TestSetup
+			var target string
+			oldSecretData := map[string][]byte{
+				"a": []byte("b"),
+			}
+			newSecretData := map[string][]byte{
+				"x": []byte("foo"),
+			}
+			sd := remotesecretstorage.SecretData(oldSecretData)
+			uploadSecret := &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "test-remote-secret-upload",
+					Namespace:   "default",
+					Labels:      map[string]string{api.UploadSecretLabel: "remotesecret"},
+					Annotations: map[string]string{api.RemoteSecretNameAnnotation: "test-remote-secret"},
+				},
+				Data: newSecretData,
+			}
 
-	})
+			BeforeEach(func() {
+				target = string(uuid.NewUUID())
+				Expect(ITest.Client.Create(ITest.Context, &v1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{Name: target},
+				})).To(Succeed())
 
+				test = crenv.TestSetup{
+					ToCreate: []client.Object{
+						&api.RemoteSecret{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "test-remote-secret",
+								Namespace: "default",
+							},
+							Spec: api.RemoteSecretSpec{
+								Secret: api.LinkableSecretSpec{
+									Name: "injected-secret",
+								},
+								Targets: []api.RemoteSecretTarget{{
+									Namespace: target,
+								}},
+							},
+						},
+					},
+					MonitoredObjectTypes: []client.Object{
+						&v1.Secret{},
+					},
+				}
+
+				test.BeforeEach(ITest.Context, ITest.Client, nil)
+				rs := *crenv.First[*api.RemoteSecret](&test.InCluster)
+				Expect(rs).NotTo(BeNil())
+				Expect(ITest.Storage.Store(ITest.Context, rs, &sd)).To(Succeed())
+			})
+			AfterEach(func() {
+				test.AfterEach(ITest.Context)
+			})
+
+			It("updates the target secret with new data", func() {
+				// check that target secret is created in each target
+				test.ReconcileWithCluster(ITest.Context, func(g Gomega) {
+					var secret = &v1.Secret{}
```

Comment:
Because secrets are monitored in this test, I think they should end up in `test.InCluster`. That means you should be able to use `crenv.Find[*v1.Secret](client.ObjectKey{Name: "injected-secret", Namespace: target})`.